### PR TITLE
[Docs] update Japanese translation of internals_gpio_control.md

### PR DESCRIPTION
**Pull request by mtei**
## Description

Update Japanese translation of docs/internals_gpio_control.md.

 * Add 'Atomic Operation' section
 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by mtei** [comment]
@umi-umi @shelaf @s-show @hsgw @satt99 @takashiski @yoshihara よかったらレビューお願いします。

**Review by shelaf — APPROVED** [review]

File: docs/ja/internals_gpio_control.md [hunk 1]
```diff
@@ -26,3 +26,22 @@ QMK には、マイクロコントローラに依存しない GPIO 制御抽象
...
 各マイクロコントローラは GPIO に関して複数の高度な設定を持つことができます。この抽象レイヤーは、アーキテクチャー固有の機能の使用法を制限しません。上級ユーザは、目的のデバイスのデータシートを参照し、必要なライブラリを含めてください。AVR については、標準 avr/io.h ライブラリが使われます; STM32 については ChibiOS [PAL ライブラリ](https://chibios.sourceforge.net/docs3/hal/group___p_a_l.html)が使われます。
+
+## アトミック操作 :id=atomic-operation
+
+上記の関数は、必ずしもアトミックに動作することが保証されているわけではありません。そのため、上記の関数を複数組み合わせて使用する際に、操作の途中での割り込みを防ぎたい場合は、以下の `ATOMIC_BLOCK_FORCEON` マクロを使用してください。
+
+例:
+```c
+void some_function() {
+     // some process
```
Comment (s-show) [inline docs/ja/internals_gpio_control.md:37]: ソースコードのコメントも翻訳されますか？
```suggestion
     // いくつかの処理
```

Comment (mtei) [inline docs/ja/internals_gpio_control.md:37]:
```suggestion
     // 通常の処理
```

File: docs/ja/internals_gpio_control.md [hunk 2]
```diff
@@ -26,3 +26,22 @@ QMK には、マイクロコントローラに依存しない GPIO 制御抽象
...
+     ATOMIC_BLOCK_FORCEON {
+        // Atomic Processing
```
Comment (mtei) [inline docs/ja/internals_gpio_control.md:39]:
```suggestion
        // アトミックであることが必要な処理
```

File: docs/ja/internals_gpio_control.md [hunk 3]
```diff
@@ -26,3 +26,22 @@ QMK には、マイクロコントローラに依存しない GPIO 制御抽象
...
+     }
+     // some process
```
Comment (mtei) [inline docs/ja/internals_gpio_control.md:41]:
```suggestion
     // 通常の処理
```

File: docs/ja/internals_gpio_control.md (hunk 1 above)
Comment (mtei) [reply docs/ja/internals_gpio_control.md]: 直してみました。

**Review by s-show — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**